Sources/SKSupport/AsyncUtils.swift

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public extension Task {
+  /// Awaits the value of the result.
+  ///
+  /// If the current task is cancelled, this will cancel the subtask as well.
+  var valuePropagatingCancellation: Success {
+    get async throws {
+      try await withTaskCancellationHandler {
+        return try await self.value
+      } onCancel: {
+        self.cancel()
+      }
+    }
+  }
+}
+
+/// Allows the execution of a cancellable operation that returns the results
+/// via a completion handler.
+///
+/// `operation` must invoke the continuation's `resume` method exactly once.
+///
+/// If the task executing `withCancellableCheckedThrowingContinuation` gets
+/// cancelled, `cancel` is invoked with the handle that `operation` provided.
+public func withCancellableCheckedThrowingContinuation<Handle, Result>(
+  _ operation: (_ continuation: CheckedContinuation<Result, any Error>) -> Handle,
+  cancel: (Handle) -> Void
```

Comment:
I think this probably ought to be marked `@Sendable` (I believe we'll warn under strict concurrency)

Reply:
Talking of strict concurrency checking, are we planning on enabling that for this repo? I don't _think_ it's a proper package option yet, but I think you can add `.enableExperimentalFeature("StrictConcurrency")` as a build setting

Reply:
I was planning to try and enable strict concurrency checking once I don’t have any further async migrations in my backlog. Until I do that, I don’t think I care about sendable annotations too much.

Reply:
Fair enough